Consolidate the list of static analyzer checkers and projects in webkitpy.safer_cpp.checkers

rniwa opened the pull request:
#### b934adffd69c5846bbb27994a4f9e423be01c044
<pre>
Consolidate the list of static analyzer checkers and projects in webkitpy.safer_cpp.checkers
<a href="https://bugs.webkit.org/show_bug.cgi?id=293278">https://bugs.webkit.org/show_bug.cgi?id=293278</a>

Reviewed by Brianna Fan.

Adopted webkitpy.safer_cpp.checkers in all the safer C++ scripts so that there is exactly one
place to update the list of checkers and descriptions.

* Tools/Scripts/compare-static-analysis-results:
(compare_project_results_to_expectations):
(compare_project_results_by_run):
(compare_results):
* Tools/Scripts/generate-dirty-files:
(find_project_results):
(find_all_results):
* Tools/Scripts/webkitpy/safer_cpp/checkers.py:
(Checker):
(Checker.find_checker_by_description):

Canonical link: <a href="https://commits.webkit.org/295218@main">https://commits.webkit.org/295218@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/7521cc42f1d997b5984e03ccac478daea217dd39

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/104224 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/23928 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/14264 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/109424 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/54892 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/106264 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/24296 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/32473 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/79160 "Passed tests") | [⏳ 🧪 win-tests](https://ews-build.webkit.org/#/builders/Win-Tests-EWS "Waiting to run tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/107230 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/18879 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/94034 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/59487 "Passed tests") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/6/builds/103701 "Passed tests") | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/18691 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/12072 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/54252 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/88390 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/12131 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/111806 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/31380 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/23133 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/88183 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/31744 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/90223 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/87845 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/32734 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/10481 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/25844 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/16954 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/31309 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/36622 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/31103 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/34439 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/32663 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

briannafan reviewed (approved):
LGTM, assuming you've tested these scripts locally as a sanity check.

We could also update `Tools/Scripts/update-safer-cpp-expectations`, but that doesn't block this from landing.

rniwa committed:
Consolidate the list of static analyzer checkers and projects in webkitpy.safer_cpp.checkers
https://bugs.webkit.org/show_bug.cgi?id=293278

Reviewed by Brianna Fan.

Adopted webkitpy.safer_cpp.checkers in all the safer C++ scripts so that there is exactly one
place to update the list of checkers and descriptions.

* Tools/Scripts/compare-static-analysis-results:
(compare_project_results_to_expectations):
(compare_project_results_by_run):
(compare_results):
* Tools/Scripts/generate-dirty-files:
(find_project_results):
(find_all_results):
* Tools/Scripts/webkitpy/safer_cpp/checkers.py:
(Checker):
(Checker.find_checker_by_description):

Canonical link: https://commits.webkit.org/295218@main
